stubs/pyserial/serial/tools/miniterm.pyi

```diff
@@ -1,18 +1,27 @@
 import codecs
 import sys
 import threading
-from _typeshed import Unused
+from _typeshed import SupportsFlush, SupportsWrite, Unused
 from collections.abc import Iterable
-from typing import Any, BinaryIO, TextIO
+from typing import Any, Protocol, TypeVar, type_check_only
 from typing_extensions import Self
 
 from serial import Serial
 
+_AnyStr_T = TypeVar("_AnyStr_T", contravariant=True)
+
+@type_check_only
+class _SupportsWriteAndFlush(SupportsWrite[_AnyStr_T], SupportsFlush, Protocol): ...
+
+@type_check_only
+class _SupportsRead(Protocol):
+    def read(self, n: int, /) -> str: ...
+
 def key_description(character: str) -> str: ...
 
 class ConsoleBase:
-    byte_output: BinaryIO
-    output: codecs.StreamWriter | TextIO
+    byte_output: _SupportsWriteAndFlush[bytes]
+    output: _SupportsWriteAndFlush[str]
```

Comment:
The one concern I have with this, is that the user looses the ability to use any other stdout methods, like `writelines` for a simple example.
```py
>>> from serial.tools.miniterm import *
>>> ConsoleBase().output.writelines("aaaa\n")
aaaa
>>>
```

I am not familiar with pyserial, but is that intentional with this change?

Reply:
We can always make this generic over the output types if a user complains about this. But at the moment I don't see why a user wouldn't just use stdout/stderr directly.